sympy/polys/orderings.py

```diff
@@ -225,7 +225,7 @@ def monomial_key(order=None, gens=None):
     if isinstance(order, Symbol):
         order = str(order)
 
-    if isinstance(order, str):
+    if isinstance(order, string_types):
```

Comment:
This is not necessary. All accepted order names (as in `__all__` above) only contain ascii letters.